file-service-persistence/src/main/java/uk/gov/justice/services/fileservice/repository/ContentJdbcRepository.java

```diff
@@ -74,37 +83,33 @@ public void update(final UUID fileId, final byte[] content, final Connection con
             resultSet = preparedStatement.executeQuery();
 
             if (resultSet.next()) {
-                return of(resultSet.getBytes(1));
+                return of(resultSet.getBinaryStream(1));
             }
         } catch (final SQLException e) {
-            throw new TransactionFailedException(format("Exception while reading metadata %s", fileId), e);
+            throw new FileServiceException(format("Exception while reading metadata %s", fileId), e);
         } finally {
             closer.close(resultSet, preparedStatement);
```

Comment:
Is it safe to close the result set before the file content stream has been read?